src/region_labelling.rs

```diff
@@ -241,6 +238,102 @@ where
 
     out
 }
+/// Given a seed point to start at and a new color to fill the region with,
+/// fill all connected pixels matching the original color of the seed point with the
+/// new color
+///
+/// ```
+/// # extern crate image;
+/// # #[macro_use]
+/// # extern crate imageproc;
+/// # fn main() {
+///
+/// use image::Luma;
+/// use imageproc::region_labelling::floodfill_mut;
+///
+/// let fill_color = Luma([5u8]);
+///
+/// let mut image = gray_image!(
+///     4, 0, 1, 1, 1;
+///     0, 1, 1, 2, 0;
+///     0, 0, 1, 0, 1;
+///     1, 0, 0, 1, 4);
+/// let seed_point = (2, 1);
+/// floodfill_mut(&mut image, &seed_point, fill_color);
+///
+/// let expected_output = gray_image!(
+///     4, 0, 5, 5, 5;
+///     0, 5, 5, 2, 0;
+///     0, 0, 5, 0, 1;
+///     1, 0, 0, 1, 4);
+///
+/// assert_pixels_eq!(image, expected_output);
+///
+/// # }
+///
+/// ```
+pub fn floodfill_mut<P,T>(image: &mut Image<P>, seed_point: &T, fill_color: P)
+where
+    P: Pixel+PartialEq,
+    T: Position
+{
+    let seed_point = (seed_point.x(), seed_point.y());
+    let (width, height) = image.dimensions();
+    let old_color = image[seed_point]; // connected pixels matching this color will be filled in with the fill_color
+    if old_color == fill_color || seed_point.0 >= width || seed_point.1 >= height {
+        return;
+    }
+    else {
+        let mut frontier = Vec::with_capacity(50); // preallocation size chosen arbitrarily as the required size varies
+        frontier.push(seed_point);
+        image[seed_point] = fill_color;
+        let connectivity = Connectivity::Four;
+        while !frontier.is_empty() {
+            let q = frontier.pop().unwrap(); // unwrap will not panic here as we have checked that frontier is not empty
+            let neighbors = neighbor_indices(image, &q, &connectivity);
+            for neighbor in neighbors {
+                if image[neighbor.into()] == old_color {
+                    let neighbor: (u32,u32) = neighbor.into();
+                    frontier.push(neighbor);
+                    image[neighbor] = fill_color;
+                }
+            }
+        }
+
+    }
+}
+
+/// Creates a clone of the input image and performs an in-place floodfill operation on 
+/// the clone using [floodfill_mut]
```

Comment:
I think references to functions in doc-comments should also use backticks like: ``[`floodfill_mut()`]``